### PR TITLE
DLRM - JAX workload Refactor

### PR DESCRIPTION
**Pull request by pomonam**
In this PR, the followings are modified:
- Refactor workload `dlrm_jax` so that we can share existing code with the upcoming PyTorch implementation (will submit PR soon).
- Update incorrect numbers in `BaseCriteo1TbDlrmSmallWorkload`. 
- Update hyperparameters and architecture to match the ones in the `init2wint` repository [here](https://github.com/google/init2winit/blob/master/init2winit/model_lib/dlrm.py).
- Incorporates pre-processing change from #103.
- Update reference submission and confirm that it passes.

TODO: We (cc' @rakshithvasudev) still need to run the workload to check if the targeted score can be achieved with this update.

### CODE REVIEW

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py [hunk 1]
```diff
@@ -219,9 +65,24 @@ def model_fn(
     del update_batch_norm
     inputs = augmented_and_preprocessed_input_batch['inputs']
     targets = augmented_and_preprocessed_input_batch['targets']
-    logits_batch = self._flax_module.apply({'params': params}, inputs, targets)
+    logits_batch = self._model.apply({'params': params}, inputs, targets)
     return logits_batch, None
 
-  @property
-  def step_hint(self):
-    return 64_000  # TODO(znado): finalize.
```
Comment (znado) [inline algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:227]: we haven't standardized it yet, but the goal for the spec.py API is to have a step_hint property on workloads that allows submitters to have a hint about how many steps it takes baseline submissions to train, so they can use it to construct LR schedules. can we keep this for now?

**Review by znado — APPROVED** [review]

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/models.py [hunk 2]
```diff
@@ -37,28 +37,17 @@ def dot_interact(concat_features):
...
-
-  Parameters:
-    vocab_sizes: list of vocab sizes of embedding tables.
-    total_vocab_sizes: sum of embedding table sizes (for jit compilation).
-    mlp_bottom_dims: dimensions of dense layers of the bottom mlp.
-    mlp_top_dims: dimensions of dense layers of the top mlp.
-    num_dense_features: number of dense features as the bottom mlp input.
-    embed_dim: embedding dimension.
-    keep_diags: whether to keep the diagonal terms in x @ x.T.
-  """
```
[the remark this answers is not in the document]
Comment (pomonam) [reply algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/models.py]: Thank you for the comment! I updated the docstring and revived it.

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py (hunk 1 above)
Comment (pomonam) [reply algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:227]: Done!

**Comment by znado** [comment]
This was fantastic, thanks so much for this! Merged into https://github.com/mlcommons/algorithmic-efficiency/pull/136

**Pull request closed without merging**